zimui/src/components/TopicHome.vue

```diff
@@ -0,0 +1,155 @@
+<script setup lang="ts">
+import TopicSection from '../components/TopicSection.vue'
+import TopicCard from '../components/TopicCard.vue'
+import { onMounted, ref, watch } from 'vue'
+
+import { useMainStore } from '../stores/main'
+const main = useMainStore()
+
+const props = defineProps({
+  slug: {
+    type: String,
+    default: undefined,
+  },
+})
+
+const data = ref()
+
+const fetchData = async function () {
+  if (props.slug == undefined) {
+    return
+  }
+  data.value = await main.fetchTopic(props.slug)
+  document.title = data.value.title
+}
+
+watch(props, fetchData)
+
+onMounted(() => {
+  fetchData()
+})
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const getTopicSections = (inputArray: any[]) => {
+  return inputArray.filter((section) => section.kind == 'topic')
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const getNonTopicSections = (inputArray: any[]) => {
+  return inputArray.filter((section) => section.kind != 'topic')
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const hasTopicAndNonTopicSection = (inputArray: any[]) => {
+  return (
+    getTopicSections(inputArray).length > 0 &&
+    getNonTopicSections(inputArray).length > 0
+  )
+}
+</script>
+
+<template>
+  <div v-if="data" class="content">
+    <div class="jumbotron" :class="{ 'with-description': data.description }">
+      <div class="container">
+        <div
+          class="align-items-start d-flex justify-content-between mt-5"
+        ></div>
+        <div class="row">
+          <div
+            :class="{
+              'col-sm-8': data.thumbnail,
+              'col-sm-12': !data.thumbnail,
+            }"
+          >
+            <router-link :to="`./${data.parents[data.parents.length - 1]}`">
```

Comment:
I'd prefer a property with an explicit name here

Reply:
I struggled to find a nice reference website where I could download them and be sure they will still here in 2 years, so I preferred to embed them in the repo to simplify maintenance.